src/main/groovy/com/pszymczyk/consul/ConsulStarter.groovy

```diff
@@ -17,7 +17,8 @@ class ConsulStarter {
 
     private final Path dataDir
     private final Path downloadDir
-    private final File portsConfigFile
+    private final Path configDir
+    private String customConfig;
```

Comment:
I think it is not necessary to store config content in this variable, maybe create config file with given content and hold refernce to this file as earlier?